stl/inc/ranges

```diff
@@ -2462,6 +2462,326 @@ namespace ranges {
+                // clang-format on
+                auto _Tmp = *this;
+                ++*this;
+                return _Tmp;
+            }
+
+            // clang-format off
+            constexpr _Iterator& operator--()
+                requires _Deref_is_ref && bidirectional_range<_Base> && bidirectional_range<_IterInnerRng>
+                      && common_range<_IterInnerRng> {
+                // clang-format on
+                if (_Outer == _RANGES end(_Parent->_Range)) {
+                    _Inner = _RANGES end(*--_Outer);
+                }
+                while (_Inner == _RANGES begin(*_Outer)) {
+                    _Inner = _RANGES end(*--_Outer);
+                }
+                --_Inner;
+                return *this;
+            }
+
+            // clang-format off
+            constexpr _Iterator& operator--(int)
+                requires _Deref_is_ref && bidirectional_range<_Base> && bidirectional_range<_IterInnerRng>
+                      && common_range<_IterInnerRng> {
+                // clang-format on
+                auto _Tmp = *this;
+                --*this;
+                return _Tmp;
+            }
+
+            // clang-format off
+            _NODISCARD friend constexpr bool operator==(const _Iterator& _Left, const _Iterator& _Right) noexcept(
+                noexcept(_Left._Outer == _Right._Outer && _Left._Inner == _Right._Inner)) /* strengthened */
+                requires _Deref_is_ref && equality_comparable<iterator_t<_Base>>
+                      && equality_comparable<iterator_t<_IterInnerRng>> {
+                // clang-format on
+#if _ITERATOR_DEBUG_LEVEL != 0
+                _Left._Same_range(_Right);
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+                return _Left._Outer == _Right._Outer && _Left._Inner == _Right._Inner;
+            }
+
+            _NODISCARD friend constexpr decltype(auto) iter_move(const _Iterator& _It) noexcept(
+                noexcept(_RANGES iter_move(_It._Inner))) {
+                return _RANGES iter_move(_It._Inner);
+            }
+
+            friend constexpr void iter_swap(const _Iterator& _Left, const _Iterator& _Right) noexcept(
+                noexcept(ranges::iter_swap(
+                    _Left._Inner, _Right._Inner))) requires indirectly_swappable<iterator_t<_IterInnerRng>> {
+                _RANGES iter_swap(_Left._Inner, _Right._Inner);
+            }
+        };
+
+        template <bool _Const>
+        class _Sentinel {
+        private:
+            template <bool>
+            friend class _Sentinel;
+
+            using _Parent_t = _Maybe_const<_Const, join_view>;
+            using _Base     = _Maybe_const<_Const, _Vw>;
+
+            sentinel_t<_Base> _Last{};
+
+            // clang-format off
+            template<bool _OtherConst> // Per resolution of LWG-3448
+                requires sentinel_for<sentinel_t<_Base>, iterator_t<_Maybe_const<_OtherConst, _Vw>>>

```

Comment:
Note to self: Use a using declaration similar to `transform_view `and `elements_view`